### ISSUE TITLE
"unable to start the service: Access is denied. (5)" trying to start service

### ISSUE THREAD

**Issue by genpfault** [TigerVNC/tigervnc#252]
tl;dr: Might want to modify the Windows installer to give "SYSTEM" read & execute permissions on `winvnc4.exe`.  Or at least add this particular edge case to the wiki.

Windows 7 Home Premium, x64

http://tigervnc.bphinz.com/nightly/
tigervnc64-1.6.80.exe
655c8ce30571de20380aa05badc4b678

Installed to `C:\Tools\VNC`

In a "Run as administrator" `cmd.exe`:

	winvnc4.exe -unregister
	(no output)

	winvnc4.exe -noconsole -start
	(popup) unable to open the service: The specified service does not exist as an installed service. (1060)
	(expected, I think)

	winvnc4.exe -register
	(no output)

	winvnc4.exe -noconsole -start
	(popup) unable to start the service: Access is denied. (5)

Looks like that's coming from here:
https://github.com/TigerVNC/tigervnc/blob/master/win/rfb_win32/Service.cxx#L534

due to `StartService()` kicking back `ERROR_ACCESS_DENIED`.

At the bottom of [Microsoft's `StartService()` docs](https://msdn.microsoft.com/en-us/library/windows/desktop/ms686321%28v=vs.85%29.aspx) "Tanuki Software" helpfully notes:

> ERROR_ACCESS_DENIED will also be returned if the account (for instance 'SYSTEM'), under who the Service will be running, doesn't have Read & Execute file permission on the Service's executable file.

My local user account was the only thing on the `winvnc4.exe` ACL according to the "Security" tab on the properties dialog.

Added the "SYSTEM" user with the "Read & execute" and "Read" permissions set and the service started right up without complaint.

**Comment by genpfault** [comment]
The work-around works on the 1.6.0 beta too:

    https://bintray.com/artifact/download/tigervnc/beta/tigervnc64-1.5.90.exe
    sha1: a35013de7b014989e7a6666f08b0ddccbe5c659c
    sha256: 6893aef6b4daebce66598604d89e21bf9ab329277ebccbf081b6960fcaf28970

**Comment by genpfault** [comment]
Maybe append `; Permissions: system-readexec` to the [`Source` entry for `winvnc4.exe`](https://github.com/TigerVNC/tigervnc/blob/master/release/tigervnc.iss.in#L23) in `tigervnc.iss.in`?

**Comment by genpfault** [comment]
Looks like `vncconfig.exe` needs the SYSTEM treatment too or else I can't get the Options panel to come up using the systray icon.

**Comment by bphinz** [comment]
Thanks for the info.  I noted an issue with the tray icon in #234 as well, but it got lost in the shuffle.

**Comment by CendioOssman** [comment]
I'm not having any of these issues on 2008R2. Is this a problem specific to the workstation versions of Windows?

**Comment by genpfault** [comment]
I may have partitioned & formatted that drive using `gparted` instead of the partitioning/formatting wizard in the Windows 7 installer.  Can't remember if the installer re-formats a NTFS partition before attempting to install or not.

It's possible that [`mkntfs`](https://en.wikipedia.org/wiki/Ntfsprogs) didn't set the same [default permissions and/or ACLs](https://helgeklein.com/blog/2012/08/windows-7-default-file-system-permissions-listing/) as the Windows 7 installer does.
